ivy.el

```diff
@@ -2226,7 +2226,7 @@ This is useful for recursive `ivy-read'."
          (dynamic-collection (ivy-state-dynamic-collection state))
          (require-match (ivy-state-require-match state))
          (caller (or (ivy-state-caller state) this-command))
-         (sort (or (ivy-state-sort state) (assoc caller ivy-sort-functions-alist)))
+         (sort (or (ivy-state-sort state) (ivy--sort-function caller)))
```

Comment:
If `caller` is not found in `ivy-sort-functions-alist`, then `ivy--sort-function` looks up the caller from the `ivy-last` state. Could that be a problem?

Reply:
Ah, at that point, ivy-last is still the previous state, not the one being reset to by ivy-reset-state. I've added a default argument so we don't use a potentially incorrect caller.